modules/local/functions.nf

```diff
@@ -61,3 +57,23 @@ def joinByMetaKeys( Map args = [:], lhs, rhs ) {
             assert args.meta in ['lhs','rhs','merge','subset']
     }
 }
+
+def assembliesFromStage( assemblies, String stage ) {
+    assemblies.filter { meta, assembly -> meta.assembly?.stage == stage }
+}
+
+def setAssemblyStage( assemblies, String stage ) {
+    assemblies.map{ meta, assemblies -> [ meta.deepMerge([ assembly: [ stage: stage, build: "${meta.assembly.assembler}-${stage}-${meta.assembly.id}" ] ]), assemblies ] }
```

Comment:
I get following error: 

N E X T F L O W  ~  version 23.10.1
WARN: It appears you have never run this project before -- Option `-resume` is ignored
Launching `/home/pippel/tests/Earth-Biogenome-Project-pilot/main.nf` [golden_yalow] DSL2 - revision: b3d3024b05
ERROR ~ Module compilation error
- file : /home/pippel/tests/Earth-Biogenome-Project-pilot/modules/local/functions.nf
- cause: The current parameter list already contains a parameter of the name assemblies
 @ line 66, column 19.
       assemblies.map{ meta, assemblies -> [ meta.deepMerge([ assembly: [ stage: stage, build: "${meta.assembly.assembler}-${stage}-${meta.assembly.id}" ] ]), assemblies ] }
      

Its probably just a typo:

```suggestion
    assemblies.map{ meta, assembly -> [ meta.deepMerge([ assembly: [ stage: stage, build: "${meta.assembly.assembler}-${stage}-${meta.assembly.id}" ] ]), assemblies ] }
```